### ISSUE TITLE
File locations are hardcoded

### ISSUE THREAD

**Issue by dandxy89** [picnicml/doddle-benchmark#3]
I'm not able to run the test locally as the files do not exist at the locations specified.

Fails with the following error:

```
[info] Running com.picnicml.doddlemodel.linear.DoddleLinearRegression 
[error] (run-main-0) java.io.FileNotFoundException: src/main/scala/com/picnicml/doddlemodel/linear/bng_pw_linear_shuffled.csv (No such file or directory)
[error] java.io.FileNotFoundException: src/main/scala/com/picnicml/doddlemodel/linear/bng_pw_linear_shuffled.csv (No such file or directory)
```

**Comment by inejc** [comment]
Yeah, this needs to be fixed. Data is not included in the repo itself. In some cases, it is created by running the python script first (e.g. by uncommenting [this](https://github.com/picnicml/doddle-benchmark/blob/master/src/main/scala/com/picnicml/doddlemodel/linear/sklearn_logistic_regression.py#L16)). All the benchmarks should definitely be self-contained.

Another thing I was thinking about is that it would probably be useful to add benchmarks with bigger datasets.

**Comment by dandxy89** [comment]
@inejc - can you add these files to the doodle-model project?

```
   linear/bng_pw_linear_shuffled.csv
   linear/mnist.csv
   linear/log_reg.csv
```

When packaging the jars the .csv files are included.

To check that files are included from the resources folder run ```jar tf doddle-model_2.12.jar ``` in ```.ivy2/local/com.picnicml/doddle-model_2.12/0.0.0-SNAPSHOT/jars```

Once the files have been added I'll start working on the other issue I've raised.

**Comment by inejc** [comment]
Should we fetch/generate those files before the benchmarks are run and cache them in a hidden dir in user's home directory? It might make sense to keep `doddle-model` jar size at minimum. This way we could use datasets of any size for benchmarking in the future.

The only file that was downloaded manually is for the logistic regression model and I'm happy to replace that. What are your thoughts on this?

**Comment by dandxy89** [comment]
I like your first idea - it’s similar to how the Keras library works. I think what they did was put the files in S3 for storage. 

For the downloading phase would it make sense to use public datasets that we could redirect and encourage the user to download manually, that way it negates the need to store and write a client to fetch the data?

**Comment by inejc** [comment]
Yeah, I think that for the initial solution this would be just fine. The easiest benchmark would probably be for the softmax classifier since MNIST data is used: http://mldata.org/repository/data/viewslug/mnist-original.

**Comment by dandxy89** [comment]
Given that we're allowing the user to generate their own datasets I'll close this issue.
